TARGET=HASWELL fails on what seems to be an AVX issue for 0.3.15

I am trying to build 0.3.15. The PKGBUILD target looks like this

```shell
make LIBNAME=libopenblasp.a NO_LAPACK=1 DYNAMIC_ARCH=0  \
       NO_SHARED=1 NO_AVX512=1 NO_CBLAS=1 NO_WARMUP=0 \
       CC=${MINGW_PREFIX}/bin/gcc                                      \
       FC=${MINGW_PREFIX}/bin/gfortran                                 \
       OPENBLAS_INCLUDE_DIR=${MINGW_PREFIX}/include/${_realname}       \
       USE_THREAD=1 NUM_THREADS=6 GEMM_MULTITHREAD_THRESHOLD=4 TARGET=HASWELL
```

The build goes through swimmingly but dies below. I don't understand why, since HASWELL has both avx and avx2. I have built OPB this way through 0.3.13 before without issue.

```shell
/mingw64/bin/gcc -c -O2 -DMS_ABI -DMAX_STACK_ALLOC=2048 -Wall -m64 -DF_INTERFACE_GFORT -DNO_LAPACK -DNO_LAPACKE -DNO_AVX512 -DSMP_SERVER -DMAX_CPU_NUMBER=6 -DMAX_PARALLEL_NUMBER=1 -DBUILD_SINGLE=1 -DBUILD_DOUBLE=1 -DBUILD_COMPLEX=1 -DBUILD_COMPLEX16=1 -DVERSION=\"0.3.15\" -UASMNAME -UASMFNAME -UNAME -UCNAME -UCHAR_NAME -UCHAR_CNAME -DASMNAME=ssymv_U -DASMFNAME=ssymv_U_ -DNAME=ssymv_U_ -DCNAME=ssymv_U -DCHAR_NAME=\"ssymv_U_\" -DCHAR_CNAME=\"ssymv_U\" -DNO_AFFINITY -I.. -UDOUBLE  -UCOMPLEX -UCOMPLEX -UDOUBLE -ULOWER ../kernel/x86_64/ssymv_U.c -o ssymv_U.obj
/mingw64/bin/gcc -c -O2 -DMS_ABI -DMAX_STACK_ALLOC=2048 -Wall -m64 -DF_INTERFACE_GFORT -DNO_LAPACK -DNO_LAPACKE -DNO_AVX512 -DSMP_SERVER -DMAX_CPU_NUMBER=6 -DMAX_PARALLEL_NUMBER=1 -DBUILD_SINGLE=1 -DBUILD_DOUBLE=1 -DBUILD_COMPLEX=1 -DBUILD_COMPLEX16=1 -DVERSION=\"0.3.15\" -UASMNAME -UASMFNAME -UNAME -UCNAME -UCHAR_NAME -UCHAR_CNAME -DASMNAME=ssymv_L -DASMFNAME=ssymv_L_ -DNAME=ssymv_L_ -DCNAME=ssymv_L -DCHAR_NAME=\"ssymv_L_\" -DCHAR_CNAME=\"ssymv_L\" -DNO_AFFINITY -I.. -UDOUBLE  -UCOMPLEX -UCOMPLEX -UDOUBLE -DLOWER ../kernel/x86_64/ssymv_L.c -o ssymv_L.obj
../kernel/x86_64/../arm/sum.c: In function 'ssum_k':
../kernel/x86_64/../arm/sum.c:72:9: warning: AVX vector return without AVX enabled changes the ABI [-Wpsabi]
   v_f32 vsum0 = v_zero_f32();
         ^~~~~
In file included from C:/rtools40/mingw64/lib/gcc/x86_64-w64-mingw32/8.3.0/include/immintrin.h:41,
                 from C:/rtools40/mingw64/lib/gcc/x86_64-w64-mingw32/8.3.0/include/x86intrin.h:48,
                 from C:/rtools40/mingw64/x86_64-w64-mingw32/include/winnt.h:1519,
                 from C:/rtools40/mingw64/x86_64-w64-mingw32/include/minwindef.h:163,
                 from C:/rtools40/mingw64/x86_64-w64-mingw32/include/windef.h:8,
                 from C:/rtools40/mingw64/x86_64-w64-mingw32/include/windows.h:69,
                 from ../common.h:119,
                 from ../kernel/x86_64/../arm/sum.c:32:
../kernel/x86_64/../arm/../simd/intrin_avx.h: In function 'v_sum_f32':
C:/rtools40/mingw64/lib/gcc/x86_64-w64-mingw32/8.3.0/include/avxintrin.h:516:1: error: inlining failed in call to always_inline '_mm256_extractf128_ps': target specific option mismatch
 _mm256_extractf128_ps (__m256 __X, const int __N)
 ^~~~~~~~~~~~~~~~~~~~~
In file included from ../kernel/x86_64/../arm/../simd/intrin.h:63,
                 from ../kernel/x86_64/../arm/sum.c:33:
../kernel/x86_64/../arm/../simd/intrin_avx.h:46:17: note: called from here
     __m128 hi = _mm256_extractf128_ps(sum_halves, 1);
                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from C:/rtools40/mingw64/lib/gcc/x86_64-w64-mingw32/8.3.0/include/immintrin.h:41,
                 from C:/rtools40/mingw64/lib/gcc/x86_64-w64-mingw32/8.3.0/include/x86intrin.h:48,
                 from C:/rtools40/mingw64/x86_64-w64-mingw32/include/winnt.h:1519,
                 from C:/rtools40/mingw64/x86_64-w64-mingw32/include/minwindef.h:163,
                 from C:/rtools40/mingw64/x86_64-w64-mingw32/include/windef.h:8,
                 from C:/rtools40/mingw64/x86_64-w64-mingw32/include/windows.h:69,
                 from ../common.h:119,
                 from ../kernel/x86_64/../arm/sum.c:32:
C:/rtools40/mingw64/lib/gcc/x86_64-w64-mingw32/8.3.0/include/avxintrin.h:1453:1: error: inlining failed in call to always_inline '_mm256_castps256_ps128': target specific option mismatch
 _mm256_castps256_ps128 (__m256 __A)
 ^~~~~~~~~~~~~~~~~~~~~~
In file included from ../kernel/x86_64/../arm/../simd/intrin.h:63,
                 from ../kernel/x86_64/../arm/sum.c:33:
../kernel/x86_64/../arm/../simd/intrin_avx.h:45:17: note: called from here
     __m128 lo = _mm256_castps256_ps128(sum_halves);
                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from C:/rtools40/mingw64/lib/gcc/x86_64-w64-mingw32/8.3.0/include/immintrin.h:41,
                 from C:/rtools40/mingw64/lib/gcc/x86_64-w64-mingw32/8.3.0/include/x86intrin.h:48,
                 from C:/rtools40/mingw64/x86_64-w64-mingw32/include/winnt.h:1519,
                 from C:/rtools40/mingw64/x86_64-w64-mingw32/include/minwindef.h:163,
                 from C:/rtools40/mingw64/x86_64-w64-mingw32/include/windef.h:8,
                 from C:/rtools40/mingw64/x86_64-w64-mingw32/include/windows.h:69,
                 from ../common.h:119,
                 from ../kernel/x86_64/../arm/sum.c:32:
C:/rtools40/mingw64/lib/gcc/x86_64-w64-mingw32/8.3.0/include/avxintrin.h:269:1: error: inlining failed in call to always_inline '_mm256_hadd_ps': target specific option mismatch
 _mm256_hadd_ps (__m256 __X, __m256 __Y)
 ^~~~~~~~~~~~~~
In file included from ../kernel/x86_64/../arm/../simd/intrin.h:63,
                 from ../kernel/x86_64/../arm/sum.c:33:
../kernel/x86_64/../arm/../simd/intrin_avx.h:44:18: note: called from here
     sum_halves = _mm256_hadd_ps(sum_halves, sum_halves);
                  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from C:/rtools40/mingw64/lib/gcc/x86_64-w64-mingw32/8.3.0/include/immintrin.h:41,
                 from C:/rtools40/mingw64/lib/gcc/x86_64-w64-mingw32/8.3.0/include/x86intrin.h:48,
                 from C:/rtools40/mingw64/x86_64-w64-mingw32/include/winnt.h:1519,
                 from C:/rtools40/mingw64/x86_64-w64-mingw32/include/minwindef.h:163,
                 from C:/rtools40/mingw64/x86_64-w64-mingw32/include/windef.h:8,
                 from C:/rtools40/mingw64/x86_64-w64-mingw32/include/windows.h:69,
                 from ../common.h:119,
                 from ../kernel/x86_64/../arm/sum.c:32:
C:/rtools40/mingw64/lib/gcc/x86_64-w64-mingw32/8.3.0/include/avxintrin.h:269:1: error: inlining failed in call to always_inline '_mm256_hadd_ps': target specific option mismatch
 _mm256_hadd_ps (__m256 __X, __m256 __Y)
 ^~~~~~~~~~~~~~
In file included from ../kernel/x86_64/../arm/../simd/intrin.h:63,
                 from ../kernel/x86_64/../arm/sum.c:33:
../kernel/x86_64/../arm/../simd/intrin_avx.h:43:25: note: called from here
     __m256 sum_halves = _mm256_hadd_ps(a, a);
                         ^~~~~~~~~~~~~~~~~~~~
make[1]: *** [Makefile.L1:699: ssum_k.obj] Error 1
make[1]: *** Waiting for unfinished jobs....
make[1]: Leaving directory '/d/r-packages/rtools-packages/mingw-w64-openblas/src/build-x86_64/kernel'
make: *** [Makefile:170: libs] Error 1
==> ERROR: A failure occurred in build().
    Aborting...
```

Thank you.

--- a/Makefile.x86
+++ b/Makefile.x86
@@ -1,6 +1,6 @@
 # COMPILER_PREFIX = mingw32-
 
-ifndef DYNAMIC_ARCH
+ifneq ($(DYNAMIC_ARCH),1)
 ADD_CPUFLAGS = 1
 else
 ifdef TARGET_CORE

--- a/Makefile.x86_64
+++ b/Makefile.x86_64
@@ -9,7 +9,7 @@ endif
 endif
 
 
-ifndef DYNAMIC_ARCH
+ifneq ($(DYNAMIC_ARCH),1)
 ADD_CPUFLAGS = 1
 else
 ifdef TARGET_CORE

